src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -1,455 +1,38 @@
-            for intobj in self.middleware.call_sync('interface.query'):
-                if any(filter(lambda x: ip in x['address'], intobj['aliases'])):
-                    try:
-                        iindex.append(socket.if_nametoindex(intobj['name']))
-                    except OSError:
-                        self.logger.debug('Failed to determine interface index for [%s], service [%s]',
-                                          iface, service, exc_info=True)
-
-                    break
-
-        return iindex if iindex else [kDNSServiceInterfaceIndexAny]
-
-    def register(self):
-        """
-        An instance of DNSServiceRef (sdRef) represents an active connection to mdnsd.
-        """
-        mDNSServices = {}
-        for srv in ServiceType:
-            mDNSServices.update({srv.name: {}})
-            self.service = srv.name
-            self.regtype, self.port = ServiceType[self.service].value
-            if not self._is_running():
-                continue
-
-            txtrecord, interfaceIndex = self._generate_txtRecord()
-            if txtrecord is None:
-                continue
-
-            port = self._get_port()
-
-            self.logger.trace(
-                'Registering mDNS service host: %s,  regtype: %s, port: %s, interface: %s, TXTRecord: %s',
-                self.hostname, self.regtype, port, interfaceIndex, txtrecord
-            )
-            for i in interfaceIndex:
-                mDNSServices[srv.name].update({i: {
-                    'sdRef': None,
-                    'interfaceIndex': i,
-                    'regtype': self.regtype,
-                    'port': port,
-                    'txtrecord': txtrecord,
-                    'name': self.hostname
-                }})
-
-                mDNSServices[srv.name][i]['sdRef'] = pybonjour.DNSServiceRegister(
-                    name=self.hostname,
-                    regtype=self.regtype,
-                    interfaceIndex=i,
-                    port=port,
-                    txtRecord=txtrecord,
-                    callBack=None
-                )
-
-        self.finished.wait()
-        for srv in mDNSServices.keys():
-            for i in mDNSServices[srv].keys():
-                self.logger.trace('Unregistering %s %s.',
-                                  mDNSServices[srv][i]['name'], mDNSServices[srv][i]['regtype'])
-                mDNSServices[srv][i]['sdRef'].close()
-
-    def run(self):
-        set_thread_name(f'mdns_svc_{self.service}')
-        try:
-            self.register()
-        except pybonjour.BonjourError:
-            self.logger.debug("ServiceThread: failed to register '%s', is mdnsd running?", self.service)
-
-    def setup(self):
-        pass
-
-    def cancel(self):
-        self.finished.set()
+from middlewared.service import Service, private, accepts
+from middlewared.service_exception import CallError
+from middlewared.utils import run
 
 
 class mDNSAdvertiseService(Service):
```

Comment:
Do we need this service/plugin?

Could it make more sense to have _start_mdns, _reload_mdns, etc in service.py instead and use those?